test/new-e2e/pkg/environments/aws/kubernetes/kindvm.go

```diff
@@ -0,0 +1,199 @@
+	return func(params *ProvisionerParams) error {
+		params.name = name
+		return nil
+	}
+}
+
+// WithEC2VMOptions adds options to the EC2 VM
+func WithEC2VMOptions(opts ...ec2.VMOption) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.vmOptions = opts
+		return nil
+	}
+}
+
+// WithAgentOptions adds options to the agent
+func WithAgentOptions(opts ...kubernetesagentparams.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = opts
+		return nil
+	}
+}
+
+// WithFakeIntakeOptions adds options to the fake intake
+func WithFakeIntakeOptions(opts ...fakeintake.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = opts
+		return nil
+	}
+}
+
+// WithoutFakeIntake removes the fake intake
+func WithoutFakeIntake() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = nil
+		return nil
+	}
+}
+
+// WithoutAgent removes the agent
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// WithExtraConfigParams adds extra config parameters to the environment
+func WithExtraConfigParams(configMap runner.ConfigMap) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.extraConfigParams = configMap
+		return nil
+	}
+}
+
+// Provisioner creates a new provisioner
+func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.KindVMHost] {
+	params := newProvisionerParams()
+	_ = optional.ApplyOptions(params, opts)
+
+	provisioner := e2e.NewTypedPulumiProvisioner(provisionerBaseID+params.name, func(ctx *pulumi.Context, env *environments.KindVMHost) error {
+		params := newProvisionerParams()
+		_ = optional.ApplyOptions(params, opts)
+
+		return KindVMRunFunc(ctx, env, params)
+	}, params.extraConfigParams)
+
+	return provisioner
+}
+
+// KindVMRunFunc is the Pulumi run function that runs the provisioner
+func KindVMRunFunc(ctx *pulumi.Context, env *environments.KindVMHost, params *ProvisionerParams) error {
+
+	awsEnv, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return err
+	}
+
+	host, err := ec2.NewVM(awsEnv, params.name, params.vmOptions...)
+	if err != nil {
+		return err
+	}
+
+	kindCluster, err := localKubernetes.NewKindCluster(*awsEnv.CommonEnvironment, host, awsEnv.CommonNamer.ResourceName("kind"), params.name, awsEnv.KubernetesVersion())
+	if err != nil {
+		return err
+	}
+	kindCluster.Export(ctx, &env.KubernetesCluster.ClusterOutput)
+
+	kubeProvider, err := kubernetes.NewProvider(ctx, awsEnv.Namer.ResourceName("k8s-provider"), &kubernetes.ProviderArgs{
+		EnableServerSideApply: pulumi.Bool(true),
+		Kubeconfig:            kindCluster.KubeConfig,
+	})
+	if err != nil {
+		return err
+	}
+
+	if params.fakeintakeOptions != nil {
+		fakeIntake, err := fakeintake.NewECSFargateInstance(awsEnv, params.name, params.fakeintakeOptions...)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This option should default to "WithLoadBalancer"

Reply:
Do we want to have a Load Balancer by default for test with Fakeintake in Kindvm

Reply:
Yes, kindvm tests send a huge amount of data - see this example 
![image](https://github.com/DataDog/datadog-agent/assets/45568537/95e5f1d0-3d26-4104-acd6-7cb43872bcd1)